It's possible to cat kube-config and apply together

nicolaiskogheim opened the pull request:
If you want to programmatically edit the config files
for kubectl, then the config files needs to begin
with three dashes.

This change makes it possible to do

```sh
cat deploy/kube-config/influxdb/* \
  | sed 's/amd64/arm/g' \
  | kubectl create -f -
```

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/heapster/1535)
<!-- Reviewable:end -->

nicolaiskogheim commented:
I signed it.

piosz commented:
Does the following command work correctly:
```
kubectl create -f deploy/kube-config/influxdb/
```
with dashes?

piosz commented:
@k8s-bot ok to test

nicolaiskogheim commented:
@piosz Sorry for the delay.

> Does the following command work correctly

Yes.

AlmogBaku commented:
why not just do `kubectl apply -f /dir/`?

nicolaiskogheim commented:
@AlmogBaku Because then I cannot modify the files on the fly. Please read the first post in this thread.

piosz commented:
LGTM, please rebase.

piosz commented:
/lgtm

nicolaiskogheim committed:
It's possible to cat kube-config/influxdb and apply together

If you want to programmatically edit the config files
for kubectl, then the config files needs to begin
with three dashes.

This change makes it possible to do

cat deploy/kube-config/influxdb/*   \
| sed 's/amd64/arm/g'   \
| kubectl create -f -

nicolaiskogheim commented:
@piosz Rebased now.

I have only done this on files I use regularly, but those three dashes are a part of the YAML spec, and theres no reason not to include them. I could go through the other YAML files and add it if you want.
